pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -148,11 +148,15 @@ def interpret_operation(self, op: Operator):
                     self.previous_ops.pop(w)
                 return []
 
-            # Putting the operations in a set to avoid applying the same op multiple times
-            # Using a set causes order to no longer be guaranteed, so the new order of the
-            # operations might differ from the original order. However, this only impacts
-            # operators without any shared wires, so correctness will not be impacted.
-            previous_ops_on_wires = set(self.previous_ops.get(w) for w in op.wires)
+            seen_ops = set()
+            previous_ops_on_wires = []
+
+            for w in op.wires:
+                o = self.previous_ops.get(w)
+                if o is not None and o not in seen_ops:
+                    previous_ops_on_wires.append(o)
+                    seen_ops.add(o)
```

Comment:
So you're using a set here just so that you can do look ups with a better time complexity than a list, right?

Why not just use `dict.fromkeys` here?
```suggestion
            previous_ops_on_wires = list(dict.fromkeys(self.previous_ops.get(w) for w in op.wires))
```

Reply:
Thanks, that's also what we used for `merge_rotations` since in Python 3.8+ dicts are ordered, so this should work.